Home Work 7-1

crazydope opened the pull request:
Проверка email адресов

crazydope committed:
Home Work 7-1

nnikitos95 commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+/vendor/
+/.idea/
+composer.lock
```
почему composer.lock включили? он обычно пушится

crazydope replied:
Включил для "чистоты" работы. Валидаторы можно объединить в библиотеку или отдельный модуль (компонент) и использовать в различных проектах.
В этом случае .lock файл обычно добавляют. Примеры:
https://github.com/rakit/validation
https://github.com/Respect/Validation

nnikitos95 commented on src/ValidatorInterface.php:
```diff
@@ -0,0 +1,10 @@
+{
+    public function isValid($value): bool;
+
+    public function getErrors(): string;
```
может лучше было бы `array`?

crazydope replied:
Обычно у наших валидаторов есть 2 метода. Один, который возвращает массив, а другой, который возвращает строку. Тут реализовал строку, но переделать несложно.

nnikitos95 commented on src/AbstractValidator.php:
```diff
@@ -0,0 +1,30 @@
+
+namespace crazydope\validation;
+
+abstract class AbstractValidator
```
Зачем перенос на новую строку? в 120 символов влезает ведь

crazydope replied:
Рабочий Code Style

nnikitos95 commented on src/AbstractValidator.php:
```diff
@@ -0,0 +1,30 @@
+
+    abstract public function isValid( $value ): bool;
+
+    public function __invoke($value)
```
мне кажется, что это лишнее, в этой функции вы все равно не делаете ничего, почему просто не вызывать `isValid`? если только ради `(new HostValidator)($this->hostname)` то можно и так `(new HostValidator)->isValid($this->hostname)`

crazydope replied:
Исправлю

nnikitos95 commented on src/AbstractValidator.php:
```diff
@@ -0,0 +1,30 @@
+{
+    protected $errors = [];
+
+    protected function addError( string $message): void
```
где-то один пробел, где-то два)
Первый вопрос почему вообще так пишете, привыкли, стиль кода на работе?
Может просто делать так?  `protected function addError(string $message)`

crazydope replied:
Рабочий Code Style :)  Трудно перестроиться, да и шторм настроен. Там, где нет пробелов, мой недосмотр. Исправлю

crazydope committed:
[fix] Работа над ошибками.
1) Исправление пробелов
2) Метод getErrors возвращает array
3) Удаление метода __invoke
